src/Unpacker.php

```diff
@@ -53,7 +55,10 @@ public function unpack(Operation $op): Result
                     continue;
                 }
 
-                if (!$manipulator->addLink($package['dev'] ? 'require-dev' : 'require', $link->getTarget(), $link->getPrettyConstraint(), $op->shouldSort())) {
+                $constraint = $link->getPrettyConstraint();
+                $constraint = substr($this->resolver->parseVersion($link->getTarget(), $constraint, !$package['dev']), 1) ?: $constraint;
```

Comment:
Are we sure that removing the first char is safe in 100% of the cases?

Reply:
Yep: https://3v4l.org/29ujj